clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1228,14 +1228,17 @@ packageSYCLBIN(SYCLBIN::BundleState State,
   return *OutFileOrErr;
 }
 
+Error copyFileToFinalExecutable(StringRef File) {
+  if (std::error_code EC = sys::fs::copy_file(File, ExecutableName))
```

Comment:
Would `rename` fit better than `copy_file`?
Do we really need a copy?

@steffenlarsen, FYI.

Reply:
We can try to use 'rename'. However, I think we might have to check if -save-temps is set. If -save-temps is set, then we might have to use copy_file.

Thanks

Reply:
What is wrong with copying files between different filesystems?
Are you sure this is not allowed?

IIRC, clang-linker-wrapper is trying to delete temporary files at the exit. Can it cause the problem with deleting the renamed file?

Let's leave a TODO comment to optimize this code separately.

Reply:
I thought about that. It makes the code unnecessarily unwieldy. We can try that optimization once things stabilize a bit.

Thanks